app/views/LoginSignupView.js

```diff
@@ -329,16 +346,35 @@ class LoginSignupView extends React.Component {
 				onPress = () => this.onPressCustomOAuth(service);
 				break;
 			}
+			case 'saml': {
+				onPress = () => this.onPressSaml(service);
+				break;
+			}
+			case 'cas': {
+				onPress = () => this.onPressCas();
+				break;
+			}
 			default:
 				break;
 		}
+		name = name.charAt(0).toUpperCase() + name.slice(1);
+		let buttonText;
+		if (service.service === 'saml') {
```

Comment:
You can use `if (service.service === 'saml' || (service.service === 'cas' && this.CAS_enabled))`.

Reply:
True (and simple)! Let me change it!